libvirt/domain.go

```diff
@@ -280,17 +281,117 @@ func inlineUpdateCoreOsIgnition(d *schema.ResourceData, domainDef *libvirtxml.Do
 		if err != nil {
 			return err
 		}
-		err = guestfishExecution(rootPath, fileToUpload)
+		connURI, err := virConn.GetURI()
+		if err != nil {
+			return err
+		}
+		err = guestfishExecution(rootPath, fileToUpload, connURI)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+func getLibvirtHostFromURI(connURI string) string {
+	// example uri with all feilds: schema://userinfo@host:port/system
+	virHost := strings.Split(strings.SplitN(connURI, "//", 2)[1], "/")[0]
+	if strings.Contains(virHost, "@") {
+		virHost = strings.SplitN(virHost, "@", 2)[1]
+	}
+	virHost = strings.Split(virHost, ":")[0]
+	return virHost
+}
+
+func guestfishExecution(rootPath string, fileToUpload string, connURI string) error {
+	// If the adress is in the 127.0.0.0 block or is localhost, then it is local
+	// Another local case example: qemu+tcp:///system
```

Comment:
not sure how to interpret the second line of the comment. Do you mean this is an example that is not covered? The word "Another" is confusing here.

Reply:
Yes, the second comment line is an example that is not covered.

Reply:
Thanks. I see that you removed it completely now but i don't see any changes in the code. Please add a clearer FIXME comment here. Assuming such URLs are not treated as local, something like `FIXME: URLs of the form qemu+tcp:///system not covered & will be treated as remote`.

Reply:
It does not need a FIXME here I think, as you can see `localReg := regexp.MustCompile(`:\/\/(127\.\d+\.\d+\.\d+|localhost)|///`)`, which can identify uri that contains `///` as local. Sorry for the confusing. :)